sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -371,6 +371,39 @@ def create_share(self, **kwargs):
         except HttpResponseError as error:
             process_storage_error(error)
 
+    @distributed_trace
+    def create_if_not_exists(self, **kwargs):
+        # type: (Any) -> None
+        """Creates a new Share under the account. If a share with the
+        same name already exists, it is not changed.
+
+        :keyword Dict(str,str) metadata:
+            Name-value pairs associated with the share as metadata.
+        :keyword int quota:
+            The quota to be allotted.
+        :keyword access_tier:
+            Specifies the access tier of the share.
+            Possible values: 'TransactionOptimized', 'Hot', 'Cool'
+        :keyword access_tier: str or ~azure.storage.fileshare.models.ShareAccessTier
+
+            .. versionadded:: 12.4.0
+
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :keyword protocols:
+            Protocols to enable on the share. Only one protocol can be enabled on the share.
+        :keyword protocols: str or ~azure.storage.fileshare.ShareProtocols
+        :keyword root_squash:
+            Root squash to set on the share.
+            Only valid for NFS shares. Possible values include: 'NoRootSquash', 'RootSquash', 'AllSquash'.
+        :keyword root_squash: str or ~azure.storage.fileshare.ShareRootSquash
+        :rtype: None
```

Comment:
Similar question about return type here. `create_share` claims to return a `dict`.

Reply:
I believe same findings as previously (that `create_share` falsely claims to return a `dict`).
`create_share` returns a call to `self._client.share.create`, whose return is `None, or the result of cls(response)`.

Same as previously commented, should I include the docstring changes in this PR?